src/features/player-history-aggregator/services/player.history.service.js

```diff
@@ -1,23 +1,80 @@
 import { JSDOM } from "jsdom";
+import { TrackedPlayers } from "../../../models/tracked.players.js";
 import { Players } from "../../../models/players.js";
 
 export function addCurrentPlayersFromSteam(players, games) {
   return games.map((game, i) => {
-    game.playerHistory.push(new Players(players[i]));
+    let existingMonthAndYearIndex = getExistingMonthAndYearIndex(game);
+
+    if (existingMonthAndYearIndex === -1) {
+      game.playerHistory.push(Players.newMonthlyEntry());
+      existingMonthAndYearIndex = game.playerHistory.length - 1;
+    }
+
+    const currentMonthAndYearEntry = game.playerHistory[existingMonthAndYearIndex];
+
+    currentMonthAndYearEntry.trackedPlayers.push(new TrackedPlayers(players[i]));
```

Comment:
This part with the code in line 18 should go into a method on the players class -> `player.track(new Trac...)` and in that method you recalculate the average players once you added the new entry.